nexus/peer-postgres/src/ast.rs

```diff
@@ -1,21 +1,82 @@
 use std::ops::ControlFlow;
 
-use sqlparser::ast::{visit_relations_mut, Query};
+use sqlparser::ast::{
+    visit_relations_mut, ObjectName, ObjectType, Query, Statement, TableFactor, TableWithJoins,
+};
 
 #[derive(Default)]
 pub struct PostgresAst {
     pub peername: Option<String>,
 }
 
 impl PostgresAst {
-    pub fn rewrite(&self, query: &mut Query) {
+    pub fn rewrite_query(&self, query: &mut Query) {
         visit_relations_mut(query, |table| {
             // if the peer name is the first part of the table name,
             // remove it.
-            if Some(table.0[0].value.clone()) == self.peername {
+            if Some(table.0[0].value.clone().to_lowercase()) == self.peername {
                 table.0.remove(0);
             }
             ControlFlow::<()>::Continue(())
         });
     }
+
+    #[tracing::instrument(name = "peer_postgres::rewrite_statement", skip(self))]
+    pub fn rewrite_statement(&self, stmt: &mut Statement) -> anyhow::Result<()> {
+        /* could be written less clunkily when https://github.com/rust-lang/rust/issues/53667 is stabilized.
+        Tracking in #225 */
+        let table_name: &mut ObjectName = if let Statement::Insert {
```

Comment:
can we use the visitor pattern to simplify a lot of these for example we seem to be working with TableFactor::Table in multiple branches.